packages/react-bootstrap-table2/test/row-selection/selection-cell.test.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import SelectionCell from '../../src/row-selection/selection-cell';
+
+describe('<SelectionCell />', () => {
+  const mode = 'checkbox';
+
+  let wrapper;
+
+  describe('shouldComponentUpdate', () => {
+    const selected = true;
+
+    describe('when selected prop has not been changed', () => {
+      it('should not update component', () => {
+        const nextProps = { selected };
+
+        wrapper = shallow(<SelectionCell rowKey={1} mode={mode} selected={selected} />);
+
+        expect(wrapper.instance().shouldComponentUpdate(nextProps)).toBe(false);
+      });
+    });
+
+    describe('when selected prop has been changed', () => {
+      it('should update component', () => {
+        const nextProps = { selected: !selected };
+
+        wrapper = shallow(<SelectionCell rowKey={1} mode={mode} selected={selected} />);
+
+        expect(wrapper.instance().shouldComponentUpdate(nextProps)).toBe(true);
+      });
+    });
+  });
+
+  describe('handleRowClick', () => {
+    let spy;
+
+    const mockOnRowSelect = jest.fn();
+
+    beforeEach(() => {
+      spy = jest.spyOn(SelectionCell.prototype, 'handleRowClick');
+
+      wrapper = shallow(
+        <SelectionCell
+          selected
+          rowKey={1}
+          mode={mode}
+          onRowSelect={mockOnRowSelect}
+        />
+      );
+    });
+
+    describe('when input was been clicked', () => {
+      it('should call handleRowClicked', () => {
+        wrapper.find('td').simulate('click');
+
+        expect(spy).toHaveBeenCalled();
+        expect(mockOnRowSelect).toHaveBeenCalled();
```

Comment:
How do you know this function was calling with correct value?? as I know onRowSelect will be call with `rowKey` and `checked`. I means you only check  if this function is called but doesn't check argument

Reply:
In addition `const mockOnRowSelect = jest.fn();`

I think we should have a unified solution for mocking function, I use `sinon` in this project, but if `jest`  can do same as `sinon` and better than `sinon`, I think we can change to `sinon`, or not, we should use `sinon`.

Reply:
First, I don't give the params a check `coz I thought the responsibility of this test case is to guarantee the `handleRowClick` and callback have been called. Besides, it's difficult to test :( I take a alternative way to handle logical part in test `bootstrap-table-test#handleRowSelect`.

Second, I thought `jest.fn()` maybe be better `coz it won't rely on another third party dependency and they're almost the same. However, it's up to you :)

Reply:
> First, I don't give the params a check 'coz I thought the responsibility of this test case is to guarantee the handleRowClick and callback have been called

so... you can write another test case to test `mockOnRowSelect` or mixing then in this test case



> Besides, it's difficult to test :(

I dont think so, I can test it easy, you can check my PR before.



> Second, I thought jest.fn() may be better `coz it won't rely on another third party dependency

sure, but my concern is it can do as same as `sinon`? even or better than `sinon`?

Reply:
> I dont think so, I can test it easy, you can check my PR before.

Sorry I was wrong. It could be tested easily. I'll fix that. Thanks :1st_place_medal: